lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2051,22 +2053,17 @@ data ErrWriteTxEra
     -- Exposing a 'Set AnyCardanoEra' of the candidate tx eras /could/ be done,
     -- but would require some work.
     deriving (Show, Eq)
+    deriving anyclass (Exception)
 
 readNodeTipStateForTxWrite
     :: NetworkLayer IO block
     -> IO (Write.InAnyRecentEra Write.ProtocolParameters, TimeTranslation)
 readNodeTipStateForTxWrite netLayer = do
     timeTranslation <- toTimeTranslation (timeInterpreter netLayer)
-
-    res <- currentLedgerProtocolParameters
-        <$> currentProtocolParameters netLayer
-
-    case Write.toRecentEraGADT res of
+    mpp <- currentProtocolParametersInRecentEras netLayer
+    case toRecentEraGADT mpp of
+        Left nopp -> throwIO $ ErrNodeNotYetInRecentEra nopp
         Right pp -> pure (pp, timeTranslation)
-        Left era -> throwIO $ invalidEra era
-  where
-    invalidEra =
-        ExceptionWriteTxEra . ErrNodeNotYetInRecentEra
```

Comment:
I believe the wrapping `ExceptionWriteTxEra` is required to make the error render correctly in the API and should be preserved, as we only catch exceptions of the exact type`WalletException` https://github.com/cardano-foundation/cardano-wallet/blob/6c0f7bda06a8b9649872d13311e9a0b097a4fc7c/lib/wallet/api/http/Cardano/Wallet/Shelley.hs#L448

Reply:
(In the future, we could consider using exception subhierarchies instead)